### ISSUE TITLE
bintray.com blocked in Iran

### ISSUE THREAD

**Issue by sahandmax** [tvheadend/tvheadend-build#3]
hi
the repository 'https://bintray.com/tvheadend'
banned traffic from my country
please move to free repository

this is out of freedom software conditions

thank you

**Comment by sahandmax** [comment]
my country is iran

**Comment by dreamcat4** [comment]
@sahandmax - I am very sorry to hear that you cannot access this website from your country.

> please move to free repository

Bintray IS already a free repository @sahandmax. However we are always interested to hear about other free Debian repo hosting providers. Not for the reason you give. But for other reasons. The problem however is there seems nothing out there anywhere near as good as Bintray at the moment. To be suitable for our needs / purposes.

And it could be that the problem is more on the side of your country, than of Bintray.com itself. Because jFrog support team has been absolutely great to us. And fantastic when dealing with any technical problems / issues. They are amazing. It is without any doubt that we could not provide such a good service without them.

> banned traffic from my country

Hmm. There is also a strong implication that due to the nature of this website, that any similar alternative debian hosting site may also very likely be banned too (by your country). In other words, they are probably banning a whole category of similar websites and for similar reasons.

Therefore the only solution I can suggest to you would be to try and work around the problem. For example make use of the operating system's proxy feature (APT proxy settings), and/or tunnel through tor / or some other VPN service.

Sorry we can't help further at this time. But perhaps there are other web services and software which may be able to help you.

Closing this issue.

**Comment by adamsutton** [comment]
@sahandmax just to wade in... I somewhat agree with @dreamcat4 until we can determine whether the  blame is with bintray or something in your companies infrastructure, it's unlikely that much will happen. I completely agree though that if we were to find evidence that bintray itself were blocking iranian (or any) IPs, then I'd be quite annoyed and would certainly consider that grounds to find an alternative.

In the interim, the official bintray repo is actually being mirrored (a side-effect of how I manage the packages) @ http://apt.tvheadend.org though the structure is slightly different. I think there are details on the tvheadend wiki (see https://tvheadend.org/projects/tvheadend/wiki/AptRepository/27)

**Comment by dreamcat4** [comment]
Thank you Adam for adding that. Including the http://apt.tvheadend.org, which seems the most elegant / simple solution here (at least for the duration that it still works).

But since @sahandmax did ask about it, I decided to look a bit further than my own nose... and whadda you know! - jFrog are actually an Israeli company. Who knew? (I didnt until today). So not to want to show any kind of BIAS in this matter, I'm re-opening for further comments and discussion.

From my opinion, it is either:

* jFrog are blocking Iranian ips
* Israel are blocking Iranian ips
* Iran are blocking Israeli ips

Or 2+ combinations of those things are happening at once. Potentially any of those ^^ things are the reason. And I really do not know which is/are more likely cause of the problem. Being entirely ignorant of Iran - Israeli relations [heres a wikipedia page instead](https://en.wikipedia.org/wiki/Iran%E2%80%93Israel_relations). So you can come to your own conclusions about that.

Now just to be clear:

We do not have any political or religious motiviations in regards to making the choice of the bintray.com free service. We live in an internation world, where internet service organisations can be based in any country. And chose Bintray purely from a technical standpoint, because of the service they offer, which is also provided free for open source purposes. And that is reason we have come to use it. We do not pay jFrog any monies for their service, or ask who else it is they are being funded by. Or whether or not jFrog as an organisation chooses to find or back other political, national motivated, or religiously motivated organisations.

And also:

Depending specifically who is actually the ones blocking who, would also affect whether this specific blocking issue is Iran's fault, Isreal's fault, or jFrog's fault, or 2/3 of them, or all of them simultaneously at once. So to make a future hosting decisions without knowing that is inherently a flawed / dangerous road to start treading.

Looking at it from another perspective: 

From a user - access, and convenience / inconvenience standpoint, its far less overall impact for our users if those 1/N fraction of Iranian users in the world who are blocked can use an alternate or a proxy service. Than instead for us to stop using jFrog and force ofher installed users to have to give up access via the free bintray.com apt repo service.

Sorry I can't be of much further help ATM. But we could at least open a new ticket to jFrog support ticket and ask them to please clarify for themselves and tell us if their organisation is / is not blocking Iranian IPs. And I also encourage any affected individuals to also make their own comments / views heard about these matters. Not just to ourselves, but also to other organisations and parties which are involved.

All I know is that: If you live in Iran, then aparrently the govt blocks something like 1/4 of all popular websites. So then us trying to solve this specific issue for an Iranian user is not going to help much the broader problem here. i.e. To solve your other major internet access problems and issues you will need some effective proxy service regardless of specifically the bintray.com website.

**Comment by adamsutton** [comment]
@dreamcat4 I completely agree. Without knowing who/what is causing the blockage, moving would be pointless, we might just end up hosted somewhere else that cuts off another section of the user base.

I think leaving this issue open for now is the right thing to do. If nothing else to further the debate.

**Comment by ebraminio** [comment]
While I am unfamiliar with "tvheadend" and just found the issue on Google and actually I am just drive-by commenting here but I hope this would be useful for your decision. "75.126.118.184" is one of bintray.com IPs, located on US so two middle options of those four are unlikely to be relevant here, and this IP belongs to softlayer. Going deeper, softlayer.com site is itself also blocked on Iran and I just found [this news](http://venturebeat.com/2016/02/10/ibms-softlayer-has-quietly-blocked-users-in-iran/) which is about softlayer block of Iran.

**Comment by dreamcat4** [comment]
Thank you for those new information @ebraminio. Its appreciated.

**Comment by sahandmax** [comment]
hi to all
i think @dreamcat4 speaks is true , iran gov blocks bintray.com ip but i'm not sure becaus when i'm ping with that not response
blocked sites and addresses are forwards to filtering page of iran but this not
at all of these spease
 the mirror http://apt.tvheadend.org/ is accessable from iran but bintray.com not
please update http://apt.tvheadend.org repository for our country users
sorry for my bad english friends
thank you

**Comment by sahandmax** [comment]
here traceroute from iran
bintray.com
traceroute to bintray.com (75.126.118.184), 30 hops max, 60 byte packets
 1  _gateway (192.168.99.1)  0.340 ms  0.309 ms  0.259 ms
 2  172.23.190.49 (172.23.190.49)  2.822 ms  9.897 ms  9.885 ms
 3  172.23.190.17 (172.23.190.17)  2.731 ms  9.769 ms  9.744 ms
 4  192.168.146.209 (192.168.146.209)  9.760 ms  9.719 ms  9.700 ms
 5  192.168.144.233 (192.168.144.233)  9.622 ms *  9.560 ms
 6  192.168.0.254 (192.168.0.254)  9.451 ms * *
 7  10.10.53.93 (10.10.53.93)  22.684 ms  10.446 ms  10.428 ms
 8  10.201.146.24 (10.201.146.24)  10.368 ms  10.367 ms  10.340 ms
 9  10.202.4.94 (10.202.4.94)  10.273 ms  10.254 ms  10.354 ms
10  10.202.6.248 (10.202.6.248)  10.348 ms  10.323 ms  10.296 ms
11  172.16.1.2 (172.16.1.2)  145.477 ms  145.474 ms  137.803 ms
12  * * *
13  * * *
14  * * *
15  * * *
16  * * *
17  * * *
18  * * *
19  * * *
20  * * *
21  * * *
22  * * *
23  * * *
24  * * *
25  * * *
26  * * *
27  * * *
28  * * *
29  * * *
30  * * *

tells me goverment not banned bintray.com
bintray.com drops iranian traffic to it's servers

**Comment by Flole998** [comment]
As we are using apt.tvheadend.org this is not an issue that we can take care of. Please contact bintray if you need support with their services.
